buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -503,7 +503,12 @@ class BuildPlugin implements Plugin<Project> {
             result.rethrowFailure()
             result.assertNormalExitValue() // assert exit value in case the failure cause is null
         }
-        return (captureStdErr ? stderr : stdout).toString('UTF-8').trim()
+        String[] lines = (captureStdErr ? stderr : stdout).toString('UTF-8').trim().split('\n')
+        if (lines.length == 0) {
+            throw new GradleException("Expected command to produce output but it did not.")
+        }
+
+        return lines[lines.length - 1]
```

Comment:
Why only the last line? Why do we need to split instead of just checking the length of the entire output being zero?

Reply:
Because the JVM logs warning messages to stderr. What was happening is Jenkins sets `JAVA_TOOL_OPTIONS` which then results in `jshell` printing `Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8` to stderr. We need to ignore any such output and only care about the last thing printed out, which is going to be our script output.